framework/src/modules/dpos_v2/commands/pom.ts

```diff
@@ -140,40 +143,19 @@ export class ReportDelegateMisbehaviorCommand extends BaseCommand {
 			throw new Error('Cannot apply proof-of-misbehavior. Delegate is already punished.');
 		}
 
-		const { generatorKey: generatorPublicKey1 } = await this._validatorsAPI.getValidatorAccount(
+		if (!header1.generatorAddress.equals(header2.generatorAddress)) {
```

Comment:
this never happens by the contradiction rule, but to make sure that below function make sense. (ie: below function is using the same generator key for both headers